watchdog/src/endpoints.rs

```diff
@@ -0,0 +1,353 @@
+use crate::http::HttpRequestConfig;
+use crate::print;
+use crate::{
+    transform_api_bitaps_com_block, transform_api_blockchair_com_block,
+    transform_api_blockcypher_com_block, transform_bitcoin_canister,
+    transform_blockchain_info_hash, transform_blockchain_info_height,
+    transform_blockstream_info_hash, transform_blockstream_info_height,
+    transform_chain_api_btc_com_block,
+};
+use ic_cdk::api::management_canister::http_request::{HttpResponse, TransformArgs};
+use regex::Regex;
+use serde_json::json;
+
+/// Bitcoin Block API endpoints.
+/// Allows to store the configuration of each endpoint, which includes:
+/// - URL
+/// - Transform function canister endpoint
+/// - Transform function inner implementation to be called inside the canister's endpoint
+#[derive(Debug)]
+pub enum Endpoint {
```

Comment:
1. I wanted to move from separate modules/files to just one for all the endpoints. 
    - Because there are not so many (5-6)
    - Those are unique instances of `HttpRequestConfig` that take just a few input values (url, exposed method, inner transform implementation function)
    - I suppose it improves readability to have all the constructors and tests next to each other (since they look very much the same with slight differences)
  
2. Enum was a good enough choice, since it provides a mapping of a unique name to a specific `HttpRequestConfig`. Its URL is also unique, but it's a bit too long and it's a string -- prone to typo-errors.
    - it can also be a number of free functions, then the function name would be a unique config name.
    
what do you think about switching to just functions? Eg.
```rust
endpoint_api_bitaps_com_block()
endpoint_api_blockchair_com_block()
endpoint_api_blockcypher_com_block()
endpoint_bitcoin_canister()
endpoint_blockchain_info_hash()
endpoint_blockchain_info_height()
endpoint_blockstream_info_hash()
endpoint_blockstream_info_height()
endpoint_chain_api_btc_com_block()
```

Reply:
switched to just functions, PTAL.